[[JavaBase]]
class Demo{
%%Статический метод%%
	static void generator(){
%%Контролируемый код%%
		try{
%%Создание объекта исключения%%
			NullPointerException obj=new NullPointerException(
			"Наша Ошибка"
			);
%%Генерирование исключения%%
			throw obj;
%%Обработка исключения%%
		}catch(NullPointerException e){
			System.out.println("Первый перехват исключения:");
			System.out.println("<"+e+">");
%%Повторное генерирование исключения %%
			throw e;
		}
	}
%%Главный метод%%
	public static void main(String[] args){
%%Контролируемый код%%
		try{
			generator();
%%Обработка исключения%%
		}catch(NullPointerException e){
			System.out.println("Повторный перехват:");
			System.out.println("{"+e+"}");
		}
		System.out.println("Выполнение программы завершено");
	}
}

Результат выполнения программы

Первый перехват исключения:
<java.lang.NullPointerException: Наша Ошибка>
Повторный перехват:
{java.lang.NullPointerException: Наша Ошибка}
Выполнение программы завершено